packer/table_transformations/csr_transformations.py

```diff
@@ -60,29 +65,38 @@ def transform_obs_df(df: DataFrame) -> DataFrame:
         non_study_columns = [c for c in ID_COLUMN_MAPPING.keys() - [SUBJECT_ID_FIELD, 'Study'] if c in df.columns]
         study_df.drop(columns=non_study_columns, inplace=True)
     df = from_obs_df_to_csr_df(sample_df)
-    df = merge_non_hierarchical_entity_df(df, radiology_df, 'Radiology Id', ['Subject Id', 'Diagnosis Id'])
-    df = merge_non_hierarchical_entity_df(df, study_df, 'Study Id', ['Subject Id'])
+
+    if radiology_df is not None:
+        empty_diagnosis_in_sample_df = 'Diagnosis Id' in df.index.names \
```

Comment:
I don't really understand this part. If I comment it out, a test fails (`test_result_data_shape_patient_radiology_with_diagnosis`): 
```
DataFrame.index levels are different
[left]:  3, MultiIndex([('P1',   '', 'R1'),
            ('P2',   '', 'R3'),
            ('P1', 'D2', 'R2')],
           names=['Subject Id', 'Diagnosis Id', 'Radiology Id'])
[right]: 2, MultiIndex([('P1', 'R1'),
            ('P1', 'R2'),
            ('P2', 'R3')],
           names=['Subject Id', 'Radiology Id'])
```
Looks like in the test data there should be a diagnosis identifier, right? Even though there may be no additional info from the sample df. But I may be missing something here.

Reply:
This was the tricky part that caused issues in Alessia's tests.
In principle, if variables of an entity are not selected for export in GB, the ID of this entity is not included as a column in export. 
For diagnosis entity it's different, because even though no diagnosis variable was selected to be included in export (for this test it was only Patient.sex and Radiology.body_part), the Diagnosis_Id column still ends up in data from transmart, because it is linked to Radiology. 
This was causing issues on merging sample_df and radiology_df (where merge is on both Subject Id and Diagnosis Id) - on the sample_df side all diagnosis ids were empty strings (because no diagnosis variable selected), on radiology_df it was not that case. Apparently it is difficult to make data frame treat empty strings as a match for any non-empty string (I even tried replacing them with Nan, but then has some other issues with value types).
This is why I decided to remove the diagnosis if from the export, if diagnosis variables are not selected.

Reply:
Wow, great analysis. I now understand, I think ;)

Reply:
I just included some additional comments in the code to explain this, but it's still a bit complex...